src/pybind/mgr/volumes/fs/async_job.py

```diff
@@ -27,13 +27,19 @@ def run(self):
         thread_id = threading.currentThread()
         assert isinstance(thread_id, JobThread)
         thread_name = thread_id.getName()
+        log.debug("thread [{0}] starting".format(thread_name))
 
         while retries < JobThread.MAX_RETRIES_ON_EXCEPTION:
             vol_job = None
             try:
                 # fetch next job to execute
                 with self.async_job.lock:
                     while True:
+                        # handle reconfigure of cloner_threads
+                        if len(self.async_job.threads) > self.async_job.nr_concurrent_jobs:
```

Comment:
nit: introduce a helper in AsyncJob class for this.

Reply:
Done.